AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects and behavior used in testing the script canvas variable manager
+"""
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtCore
+import pyside_utils
+from types import SimpleNamespace
+from consts.scripting import (VARIABLE_MANAGER_QT, VARIABLE_PALETTE_QT, ADD_BUTTON_QT, GRAPH_VARIABLES_QT)
+from consts.general import (WAIT_TIME_SEC_3)
+
+
+class Tests():
+    variable_count_expected = ("Variable count matches expected", "Variable count does not match expected")
+
+
+"""
+Basic variable types.
+These are str-str kv pairings so they can be used with SimpleNamespace class easily or as an iterable type.  
+"""
+VARIABLE_TYPES_DICT = {
+    "Boolean": "Boolean",
+    "Color": "Color",
+    "EntityId": "EntityId",
+    "Number": "Number",
+    "String": "String",
+    "Transform": "Transform",
+    "Vector2": "Vector2",
+    "Vector3": "Vector3",
+    "Vector4": "Vector4"
+}
+
+
+class QtPyScriptCanvasVariableManager():
+    """
+    QtPy class for handling the behavior of the script canvas variable manager
+    """
+
+    def __init__(self, sc_editor):
+
+        self.variable_manager = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, VARIABLE_MANAGER_QT)
+        self.variable_types = self.get_basic_variable_types()
+
+    def __validate_new_variable(self, new_variable_type: str) -> None:
+        """
+        function for checking the provided variable type for validity
+
+        returns: None
+        """
+        assert type(new_variable_type) is str, "Wrong parameter type provided. new_variable_type was not str"
+        assert VARIABLE_TYPES_DICT.__contains__(new_variable_type), \
```

Comment:
Shouldn't this be has_key(new_variable_type) since `__contains__(...)` having the leading/trailing underscores denotes that the function should be treated as private?

Reply:
Looks like python 3 removed `has_key()`. However, you can use `not (new_variable_type in VARIABLE_TYPES_DICT)`

Reply:
changed this over. I want to update the error message we use but I need to look into how we format strings so I can state the type provided is not valid rather than the name of the variable.

Reply:
f strings support the f"some string, {variable}" syntax.

https://docs.python.org/3/tutorial/inputoutput.html